File: src/main/kotlin/com/witelokk/music/models/Song.kt
package com.witelokk.music.models

import com.witelokk.music.UUIDSerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.*

@Serializable
data class Song(
    @Serializable(UUIDSerializer::class) val id: UUID,
    val name: String,
    @SerialName("cover_url") val coverUrl: String?,
    @SerialName("is_favorite") val isFavorite: Boolean,
    @SerialName("duration_seconds") val durationSeconds: Int,
    val artists: List<AristSummary>,
    @SerialName("stream_url") val streamUrl: String,
)

File: internal/songs/song.go
package songs

type Song struct {
	ID              string
	Name            string
	CoverMediaID    *string
	DurationSeconds int
	StreamMediaID   string
	Artists         []ArtistSummary
}

type ArtistSummary struct {
	ID            string
	Name          string
	AvatarMediaID *string
}
